discovery/azure/azure.go

```diff
@@ -71,6 +72,9 @@ type Discovery struct {
 
 // NewDiscovery returns a new AzureDiscovery which periodically refreshes its targets.
 func NewDiscovery(cfg *config.AzureSDConfig, logger log.Logger) *Discovery {
+	if logger == nil {
+		logger = log.NewNopLogger()
```

Comment:
This is Go best practice. Passing a `nil` logger is an explicit signal that we do not want anything to be logged. The `NopLogger` default is used to not guard all calls to the logger with an `if`.